packages/migration_tool/src/PortlandLabs/Concrete5/MigrationTool/Publisher/Block/StandardPublisher.php

```diff
@@ -35,13 +38,30 @@ public function publish(Batch $batch, $bt, Page $page, $area, BlockValue $value)
             }
             // Now we import the OTHER records.
             if ($b) {
+                $app = Application::getFacadeApplication();
+                /** @var Connection $connection */
+                $connection = $app->make(Connection::class);
```

Comment:
Can this be a hard dependency of this class instead of relying on the service locator? IE:
```php
protected $db;

public function __construct(Connection $db)
{
    $this->db = $db;
}
```